### PR TITLE
Fix clipToBorder in line chart

### LINKED ISSUE #228: Clip the chart within it's border
**Is your feature request relasted to a problem? Please describe.**
I want to present series which can have huge outliers , e.g. single value skyrocketing into thousands, where majority is in the -10 to 10 range. This makes the chart unreadable and impossibile to read important details. 

See the picture:
skyrocket

**Describe the solution you'd like**
I'd like to be able to "zoom in" and clip series within borders. Currently, we can restrain Y axis but the line still goes out of bound (as well as the tooltips).
clipped

Ideally, there should also be an indicator where line goes out of bound stating the outlier value (the thousands). If possibile, the tooltips should also be restrained within chart borders.

### PR DESCRIPTION
**Pull request by onliner10**
Fixes #228 and #214

### CODE REVIEW

File: lib/src/chart/line_chart/line_chart_painter.dart [hunk 1]
```diff
@@ -139,6 +137,20 @@ class LineChartPainter extends AxisChartPainter<LineChartData>
     }
   }
 
+  void clipToBorder(ui.Canvas canvas, ui.Size size,) {
```
Comment (imaNNeoFighT) [inline lib/src/chart/line_chart/line_chart_painter.dart]: Functions defined according to the usage of them in the paint() method,
please reorder its definition.

File: lib/src/chart/line_chart/line_chart_painter.dart [hunk 2]
```diff
@@ -139,6 +137,20 @@ class LineChartPainter extends AxisChartPainter<LineChartData>
...
+    final Rect rect = Rect.fromLTRB(
+      getLeftOffsetDrawSize() - halfStrokeWidth,
+      getTopOffsetDrawSize() - halfStrokeWidth,
+      size.width -
+          (getExtraNeededHorizontalSpace() - getLeftOffsetDrawSize()) +
+          halfStrokeWidth,
+      size.height- (getExtraNeededVerticalSpace() - getTopOffsetDrawSize()) + halfStrokeWidth,
+    );
+
+    canvas.clipRect(rect, clipOp: ClipOp.intersect);
```
Comment (imaNNeoFighT) [inline lib/src/chart/line_chart/line_chart_painter.dart]: The default value of `clipOp` is `ClipOp.intersect`, then you don't have to to define it.

**Comment by imaNNeoFighT** [comment]
And thank you, I didn't know that we have a `canvas.clipRect()` function.
<3
please change the target branch to `dev`

File: lib/src/chart/line_chart/line_chart_painter.dart (hunk 2 above)
Comment (onliner10) [reply lib/src/chart/line_chart/line_chart_painter.dart]: Fixed

File: lib/src/chart/line_chart/line_chart_painter.dart (hunk 1 above)
Comment (onliner10) [reply lib/src/chart/line_chart/line_chart_painter.dart]: I don't get it, it's the first function invoked within paint method and it's also declared right after paint

**Comment by onliner10** [comment]
@imaNNeoFighT Rebased onto dev. Could you please look again?

File: lib/src/chart/line_chart/line_chart_painter.dart (hunk 1 above)
Comment (imaNNeoFighT) [reply lib/src/chart/line_chart/line_chart_painter.dart]: You're right.

**Pull request merged**